src/components/application_manager/src/rpc_handler_impl.cc

```diff
@@ -449,6 +445,37 @@ bool RPCHandlerImpl::ConvertMessageToSO(
   return true;
 }
 
+bool RPCHandlerImpl::ValidateRPCSO(smart_objects::SmartObject* message,
+                                   uint32_t connection_key,
+                                   uint32_t correlation_id,
+                                   int32_t function_id,
+                                   utils::SemanticVersion& msg_version,
+                                   bool remove_unknown_params) {
+  rpc::ValidationReport report("RPC");
+  if (!message ||
+      !mobile_so_factory().attachSchema(
+          *message, remove_unknown_params, msg_version) ||
+      ((message->validate(&report, msg_version) !=
+        smart_objects::errors::OK))) {
+    LOG4CXX_WARN(logger_,
+                 "Failed to parse string to smart object with API version "
+                     << msg_version.toString() << " : " << message->asString());
+
+    std::shared_ptr<smart_objects::SmartObject> response(
```

Comment:
I think this might be better outside of this function, as it doesn't match the naming of the function. I will note that this would require the `report` object to be an added as an output parameter, though it would remove the need for other message-related parameters.

Reply:
This would make sense if we wanted to have a different response(or no response) every time we used this function. But since its a generic error related to validating the RPC smart object shouldn't we just let it remain in this function?

Reply:
The way that the function is worded, it implies that it performs the validation and returns the result, not that it sends a response within the actual function. Unfortunately, I can't think of a better name for the function which would encompass this behavior, but if it was renamed properly I'd be fine with it. I could also see a case where you might just want to not to send a response back (as is done with HMI notifications at the moment).

Reply:
Moved the response message outside of the function